sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -97,20 +97,30 @@ case class UnaryPositive(child: Expression)
 case class Abs(child: Expression)
     extends UnaryExpression with ExpectsInputTypes with NullIntolerant {
 
-  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(TypeCollection(NumericType, StringType))
 
-  override def dataType: DataType = child.dataType
+  override def dataType: DataType = child.dataType match {
+    case dt: NumericType => dt
+    case dt: StringType => DoubleType
+  }
 
   private lazy val numeric = TypeUtils.getNumeric(dataType)
 
-  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = dataType match {
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = child.dataType match {
     case dt: DecimalType =>
       defineCodeGen(ctx, ev, c => s"$c.abs()")
     case dt: NumericType =>
       defineCodeGen(ctx, ev, c => s"(${ctx.javaType(dt)})(java.lang.Math.abs($c))")
+    case dt: StringType =>
+      defineCodeGen(ctx, ev, c => s"java.lang.Math.abs(Double.valueOf($c.toString()))")
```

Comment:
How should it work when a given string includes non-numerical character?
e.g. https://zohoreportsapi.wiki.zoho.com/MySQL-Math-Functions.html